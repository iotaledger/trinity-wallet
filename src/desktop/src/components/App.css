@import-normalize;

@custom-media --xs (min-width: 0px); /* default */
@custom-media --sm (max-width: 480px);
@custom-media --md (max-width: 768px);
@custom-media --lg (max-width: 1024px);
@custom-media --xl (max-width: 1440px);

:root {
    --color-text: white;
    --color-yellow: #f7d002;
    --color-blue: #8bd4ff;
    --color-red: #ff6c69;
    --color-green: #9dffaf;
    --color-darkgreen: green;

    --color-label: var(--color-yellow);
    --color-button-warning: var(--color-yellow);
    --color-button-danger: var(--color-red);
    --color-button-success: var(--color-green);
    --color-button-extra: var(--color-blue);
    --color-button-cta: var(--color-darkgreen);
    --color-button-default: white;

    /*--font-size-default: 1.8vh;*/
    --font-size-default: 14px;
    --font-size-small: 12px;
    --font-size-title: 20px;

    --z-index-modal: 2048;
    --z-index-overlay: 1024;
    --z-index-dropdown: 512;
    --z-index-elevated: 1;
}

* {
    &,
    &:before,
    &:after {
        box-sizing: border-box;
    }
}

html,
body {
    margin: 0;
    padding: 0;
    height: 100%;
}

body {
    background-image: url(../../../shared/images/bg-green.jpg);
    background-repeat: no-repeat;
    background-size: 100% 100%;
    background-attachment: fixed;
    color: var(--color-text);
    font-family: arial;
    font-size: var(--font-size-default);
    line-height: 1.5;
}

h1 {
    font-size: var(--font-size-title);
}

label {
    color: var(--color-label);
    display: block;
    margin-bottom: 8px;
}

select {
    &:focus {
        outline: 0;
    }
    background: rgba(0, 0, 0, 0.2);
    border: 0;
    padding: 16px 4px;
    width: 100%;
    border-bottom: 2px solid white;
    color: white;
}

option {
    background: transparent;
    color: black;
    padding: 4px;
}

input {
    box-sizing: border-box;

    &:focus {
        outline: none;
    }

    &[type='text'],
    &[type='email'],
    &[type='tel'],
    &[type='password'] {
        background: rgba(0, 0, 0, 0.2);
        border: 0;
        border-bottom: 2px solid white;
        color: white;
        padding: 16px;
        width: 100%;
    }
}

textarea {
    background: rgba(0, 0, 0, 0.2);
    border: 0;
    border-bottom: 2px solid white;
    box-sizing: border-box;
    color: white;
    padding: 16px;
    width: 100%;
    &:focus {
        outline: none;
    }
}
