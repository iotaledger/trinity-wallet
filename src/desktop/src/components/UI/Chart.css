.chart {
    width: 100%;
    overflow: hidden;

    nav,
    ul {
        width: 100%;
        display: flex;
        justify-content: space-between;
    }
    nav {
        button,
        p {
            margin: 0px;
        }
        p {
            line-height: 21px;
            font-size: 21px;
            text-align: center;
        }
        button {
            display: block;
            span {
                margin-right: 6px;
            }
        }
    }
    ul {
        margin: 0px;
        padding: 0px;
        li {
            font-size: var(--font-size-small);
            list-style: none;
        }
    }
    > div {
        width: calc(100% + 55px);
        > svg {
            width: 0px;
            height: 0px;
        }
    }
}
