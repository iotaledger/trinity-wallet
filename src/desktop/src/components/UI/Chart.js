import React from 'react';
import PropTypes from 'prop-types';
import { VictoryChart, VictoryLine, VictoryAxis, Line, VictoryLabel } from 'victory';
import { getCurrencySymbol } from 'libs/currency';
import imageChevron from 'images/chevron-down-white.png';
import Button from 'components/UI/Button';
import css from './Chart.css';

export default class Chart extends React.Component {
    static propTypes = {
        marketData: PropTypes.object.isRequired,
        setTimeframe: PropTypes.func.isRequired,
        setCurrency: PropTypes.func.isRequired,
    };

    static defaultProps = {
        marketData: {},
    };

    state = {
        price: this.props.marketData.usdPrice,
    };

    componentWillMount() {
        switch (this.props.marketData.currency) {
            case 'USD':
                this.setState({ price: this.props.marketData.usdPrice });
                break;
            case 'BTC':
                this.setState({ price: this.props.marketData.btcPrice });
                break;
            case 'ETH':
                this.setState({ price: this.props.marketData.ethPrice });
                break;
        }
    }

    getMaxY() {
        const { marketData } = this.props;
        const data = marketData.chartData[marketData.currency][marketData.timeframe];
        const maxValue = Math.max(
            ...data.map(object => {
                return object.y;
            }),
        );
        return maxValue;
    }

    getMinY() {
        const { marketData } = this.props;
        const data = marketData.chartData[marketData.currency][marketData.timeframe];
        const minValue = Math.min(
            ...data.map(object => {
                return object.y;
            }),
        );
        return minValue;
    }

    getMaxX() {
        const { marketData } = this.props;
        const data = marketData.chartData[marketData.currency][marketData.timeframe];
        const maxValue = Math.max(
            ...data.map(object => {
                return object.x;
            }),
        );
        return maxValue;
    }

    getTickValues() {
        const minValue = this.getMinY();
        const maxValue = this.getMaxY();
        return [
            minValue,
            (minValue + (minValue + maxValue) / 2) / 2,
            (minValue + maxValue) / 2,
            (maxValue + (minValue + maxValue) / 2) / 2,
            maxValue,
        ];
    }

    getPriceFormat(x) {
        const { marketData } = this.props;

        if (marketData.currency === 'USD') {
            return x.toFixed(3);
        } else if (marketData.currency === 'BTC') {
            return x.toFixed(6);
        }

        return x.toFixed(5);
    }

    timeframeFromCurrent = {
        '24h': '7d',
        '7d': '1m',
        '1m': '1h',
        '1h': '24h',
    };

    nextCurrency = {
        USD: 'BTC',
        BTC: 'ETH',
        ETH: 'USD',
    };

    changeCurrency() {
        const { marketData, setCurrency } = this.props;
        const newCurrency = this.nextCurrency[marketData.currency];
        setCurrency(newCurrency);
        this.setState({ price: marketData[`${newCurrency.toLowerCase()}Price`] });
    }

    changeTimeframe() {
        const { marketData, setTimeframe } = this.props;
        setTimeframe(this.timeframeFromCurrent[marketData.timeframe]);
    }

    render() {
        const { price } = this.state;
        const { marketData } = this.props;
        const data = marketData.chartData[marketData.currency][marketData.timeframe];
        return (
            <div className={css.chart}>
                <nav>
                    <Button variant="secondary" className="small" onClick={() => this.changeCurrency()}>
                        <img src={imageChevron} />
                        {marketData.currency}
                    </Button>
                    <p>
                        {getCurrencySymbol(marketData.currency)} {this.getPriceFormat(price)} / Mi
                    </p>
                    <Button variant="secondary" className="small" onClick={() => this.changeTimeframe()}>
                        <img src={imageChevron} />
                        {marketData.timeframe}
                    </Button>
                </nav>
                <div>
                    <VictoryChart>
                        <defs>
                            <linearGradient x1="0%" y1="0%" x2="100%" y2="0%" id="gradient">
                                <stop stopColor="#FFA25B" stopOpacity={1} offset="100%" />
                                <stop stopColor="#FFFFFF" stopOpacity={0.25} offset="0%" />
                            </linearGradient>
                        </defs>
                        <VictoryLine
                            data={data}
                            style={{
                                data: {
                                    stroke: 'url(#gradient)',
                                    strokeWidth: 1.2,
                                },
                            }}
                            domain={{
                                x: [-1, this.getMaxX() + 1],
                                y: [this.getMinY(), this.getMaxY()],
                            }}
                            scale={{ x: 'time', y: 'linear' }}
                            animate={{
                                duration: 1500,
                                onLoad: { duration: 2000 },
                            }}
                        />
                        <VictoryAxis
                            dependentAxis
                            tickFormat={x => this.getPriceFormat(x)}
                            style={{
                                axis: { stroke: 'transparent' },
                                tickLabels: { fill: 'white', fontSize: 9, fontFamily: 'Lato-Regular' },
                                ticks: { padding: 0 },
                            }}
                            gridComponent={<Line type="grid" style={{ stroke: 'white', strokeWidth: 0.1 }} />}
                            tickLabelComponent={<VictoryLabel x={0} textAnchor="start" />}
                            tickValues={this.getTickValues()}
                            domain={{
                                y: [this.getMinY(), this.getMaxY()],
                            }}
                        />
                    </VictoryChart>
                </div>
                <ul>
                    <li>MCAP: $ {marketData.mcap}</li>
                    <li>Change: {marketData.change24h}%</li>
                    <li>Volume (24h): $ {marketData.volume}</li>
                </ul>
            </div>
        );
    }
}
