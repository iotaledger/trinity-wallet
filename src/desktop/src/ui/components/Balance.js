import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getSelectedSeed } from 'selectors/seeds';
import { formatValue, formatUnit, round } from 'libs/util';
import { getCurrencySymbol } from 'libs/currency';

class Balance extends React.PureComponent {
    static propTypes = {
        seed: PropTypes.object.isRequired,
        account: PropTypes.object.isRequired,
        settings: PropTypes.object.isRequired,
        marketData: PropTypes.object.isRequired,
    };

    render() {
        const { account, settings, marketData, seed } = this.props;
        const accountInfo = account.accountInfo[seed.name];

        if (!accountInfo) {
            return null;
        }

        const currencySymbol = getCurrencySymbol(settings.currency);
        const fiatBalance = round(
            accountInfo.balance * marketData.usdPrice / 1000000 * settings.conversionRate,
        ).toFixed(2);

        return (
            <div>
                <strong>{`${formatValue(accountInfo.balance)}${formatUnit(accountInfo.balance)}`}</strong>
                <small>{`${currencySymbol} ${fiatBalance}`}</small>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    account: state.account,
    seed: getSelectedSeed(state),
    marketData: state.marketData,
    settings: state.settings,
});

export default connect(mapStateToProps)(Balance);
