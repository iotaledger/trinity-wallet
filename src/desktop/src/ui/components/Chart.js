import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { LineChart, ResponsiveContainer, Line, YAxis, Tooltip } from 'recharts';
import { format, addHours, subDays, subHours, subMinutes } from 'date-fns';

import withChartData from 'containers/components/Chart';

import Button from 'ui/components/Button';
import Icon from 'ui/components/Icon';
import css from './chart.scss';

/**
 * Chart component to display historical IOTA price charts
 */
class Chart extends PureComponent {
    static propTypes = {
        /** Current price data for selected currency */
        priceData: PropTypes.shape({
            currency: PropTypes.string.isRequired,
            symbol: PropTypes.string.isRequired,
            price: PropTypes.number.isRequired,
            volume: PropTypes.string.isRequired,
            change24h: PropTypes.string.isRequired,
            mcap: PropTypes.string.isRequired,
        }).isRequired,
        /** Chart data */
        chartData: PropTypes.shape({
            data: PropTypes.array.isRequired,
            timeframe: PropTypes.string.isRequired,
            yAxis: PropTypes.shape({
                ticks: PropTypes.array.isRequired,
            }),
        }).isRequired,
        /** Change chart currency */
        setCurrency: PropTypes.func.isRequired,
        /** Change chart time frame */
        setTimeframe: PropTypes.func.isRequired,
        /** Get current market price for a currency
         * @param {string} currency - Target currency
         */
        getPriceForCurrency: PropTypes.func.isRequired,
        /** Style price to current currency format
         * @param {number} price - Input price value for formatting
         */
        getPriceFormat: PropTypes.func.isRequired,
        /** @ignore */
        theme: PropTypes.object.isRequired,
        /** @ignore */
        t: PropTypes.func.isRequired,
    };

    renderTooltip(props) {
        if (props.active) {
            const distance = props.maxItems - props.payload[0].payload.x;

            let date = subHours(new Date(), (24 * distance) / props.maxItems);

            switch (props.timeframe) {
                case '1h':
                    date = subMinutes(new Date(), (60 * distance) / props.maxItems);
                    break;
                case '7d':
                    date = subHours(new Date(), (24 * 7 * distance) / props.maxItems);
                    break;
                case '1m':
                    date = subDays(new Date(), (30 * distance) / props.maxItems);
                    break;
                case '24h':
                    date = addHours(date, 1);
                    break;
            }

            return (
                <p className={css.label}>
                    {format(date, 'DD.MM.YY HH:mm')}
                    <br />
                    <strong>
                        {props.symbol} {props.payload[0].value}
                    </strong>
                </p>
            );
        }
    }

    render() {
        const {
            priceData,
            chartData,
            theme,
            setCurrency,
            setTimeframe,
            getPriceFormat,
            getPriceForCurrency,
            t,
        } = this.props;

        return (
            <div className={css.chart}>
                <div>
                    {chartData.data.length ? (
                        <ResponsiveContainer height="100%" width="100%">
                            <LineChart data={chartData.data}>
                                <Line
                                    strokeWidth={2}
                                    type="natural"
                                    dataKey="y"
                                    stroke={theme.chart.color}
                                    dot={false}
                                />
                                <YAxis
                                    strokeWidth={0}
                                    width={55}
                                    tickMargin={10}
                                    tick={{ fill: theme.body.color }}
                                    tickCount={6}
                                    interval={0}
                                    ticks={
                                        chartData.yAxis.ticks
                                            ? chartData.yAxis.ticks.map((tick) => getPriceFormat(tick))
                                            : null
                                    }
                                    domain={['dataMin', 'dataMax']}
                                />
                                <Tooltip
                                    timeframe={chartData.timeframe}
                                    maxItems={chartData.data.length}
                                    symbol={priceData.symbol}
                                    content={this.renderTooltip}
                                />
                            </LineChart>
                        </ResponsiveContainer>
                    ) : (
                        <Icon icon="cross" size={128} />
                    )}
                </div>
                <footer>
                    <nav>
                        <Button variant="secondary" className="outline" onClick={() => setCurrency()}>
                            {priceData.currency}
                        </Button>
                        <p>
                            {priceData.symbol} {getPriceFormat(getPriceForCurrency(priceData.currency))} / Mi
                        </p>
                        <Button variant="secondary" className="outline" onClick={() => setTimeframe()}>
                            {chartData.timeframe.replace('1m', '28d')}
                        </Button>
                    </nav>
                    <ul>
                        <li>
                            <strong>{t('chart:mcap')}: </strong>
                            {priceData.globalSymbol} {priceData.mcap}
                        </li>
                        <li>
                            <strong>{t('chart:change')}:</strong> {priceData.change24h}%
                        </li>
                        <li>
                            <strong>{t('chart:volume')}: </strong>
                            {priceData.globalSymbol} {priceData.volume}
                        </li>
                    </ul>
                </footer>
            </div>
        );
    }
}

export default withChartData(Chart);
