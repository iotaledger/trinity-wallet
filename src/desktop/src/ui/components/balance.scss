.balance {
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;

    @media (max-width: 320px) {
        height: 160px;
        padding-top: 15px;
        position: relative;
    }

    h3 {
        font-size: 14px;
        font-weight: 400;
        letter-spacing: 0.5px;
    }
    h1 {
        font-size: 64px;
        line-height: 64px;
        margin: 12px 0px;
        cursor: pointer;

        @media screen and (max-width: 860px) {
            margin: 4px 0;
        }

        small {
            font-size: 21px;
            line-height: 21px;
            margin-left: 6px;
        }
    }

    h2 {
        margin-bottom: 15px;
    }

    > a {
        position: absolute;
        top: 50%;
        left: 20px;
        margin-top: -5px;
        &:last-of-type {
            left: auto;
            right: 20px;
        }
    }
}
