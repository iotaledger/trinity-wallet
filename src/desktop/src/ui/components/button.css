@keyframes spin {
    to {
        transform: rotate(360deg);
    }
}

.button {
    position: relative;
    border: 1px solid transparent;
    border-radius: 3px;
    cursor: pointer;
    display: inline-block;
    height: 42px;
    line-height: 40px;
    font-size: var(--font-size-s);
    letter-spacing: 0.2px;
    min-width: 140px;
    text-decoration: none;
    text-transform: uppercase;
    text-align: center;
    font-weight: 200;
    padding: 0 24px 0;
    user-select: none;
    overflow: visible;
    transition: padding 0.2s cubic-bezier(0.55, 0, 0.1, 1);
    &:focus {
        outline: none;
    }
    &.small {
        padding: 0 16px;
        height: 32px;
        line-height: 32px;
        min-width: 30px;
        border-radius: 2px;
    }
    &.large,
    &.inline {
        padding: 0 32px;
        line-height: 44px;
        font-weight: 300;
        height: 46px;
        font-size: 16px;
        text-transform: capitalize;
    }
    &:disabled,
    &.disabled {
        opacity: 0.3;
        pointer-events: none;
    }
    &:before {
        content: '';
        box-sizing: border-box;
        position: absolute;
        top: 50%;
        left: 18px;
        width: 16px;
        height: 16px;
        margin: -8px 0 0 -8px;
        border-radius: 50%;
        border: 1px solid transparent;
        border-top-color: var(--color-Border);
        opacity: 0;
    }
}

footer {
    .button + .button {
        margin-left: 20px;
    }
}

.primary {
    composes: button;
    color: var(--primary-body);
    background: var(--primary);
    &:hover {
        background: var(--primary-hover);
    }
}

.secondary {
    composes: button;
    color: var(--secondary-body);
    background: var(--secondary);
    &:hover {
        background: var(--secondary-hover);
    }
}

.positive {
    composes: button;
    color: var(--positive-body);
    background: var(--positive);
    &:hover {
        background: var(--positive-hover);
    }
}

.negative {
    composes: button;
    color: var(--negative-body);
    background: var(--negative);
    &:hover {
        background: var(--negative-hover);
    }
}

.extra {
    composes: button;
    color: var(--extra-body);
    background: var(--extra);
    &:hover {
        background: var(--extra-hover);
        border-color: var(--extra-hover);
    }
}

.inline {
    background: none;
    &:hover {
        background: none;
    }
    &.secondary {
        color: var(--secondary);
        &:hover {
            color: var(--secondary-hover);
        }
    }
}

.outline {
    background: var(--body-bg);
    border: 1px solid;

    &:hover {
        background: var(--body-bg);
    }

    &.primary {
        color: var(--primary);
        &:hover {
            color: var(--primary-hover);
        }
    }

    &.secondary {
        color: var(--secondary);
        &:hover {
            color: var(--secondary-hover);
        }
    }
    &.positive {
        color: var(--positive);
        &:hover {
            color: var(--positive-hover);
        }
    }
    &.negative {
        color: var(--negative);
        &:hover {
            color: var(--negative-hover);
        }
    }
    &.extra {
        color: var(--extra);
        &:hover {
            color: var(--extra-hover);
        }
    }
}

.icon {
    border: none;
    background: none;
    text-transform: capitalize;
    font-size: 14px;
    color: var(--body);
    cursor: pointer;
    height: auto;
    padding: 0px;
    > span {
        display: block;
        cursor: pointer;
    }
    &:hover {
        background: none;
        > span {
            color: var(--primary);
        }
    }
}

.loading {
    pointer-events: none;
    padding: 0 12px 0 36px;
    &:before {
        transition: all 0.8s cubic-bezier(0.55, 0, 0.1, 1);
        opacity: 1;
        animation: spin 0.6s ease infinite;
    }
}
