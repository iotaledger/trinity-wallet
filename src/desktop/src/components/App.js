import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { persistStore } from 'redux-persist';
import { withRouter } from 'react-router-dom';
import store from 'store';
import i18next from 'libs/i18next';
import Loading from 'components/UI/Loading';
import Onboarding from 'components/Layout/Onboarding';
import Main from 'components/Layout/Main';
import Notifications from 'components/UI/Notifications';

import './App.css';

// should be used later:

// class ErrorBoundary extends React.Component {
//     static propTypes = {
//         children: PropTypes.node,
//     };
//
//     state = {};
//
//     componentDidCatch(error) {
//         this.setState(() => ({
//             error,
//         }));
//     }
//     render() {
//         if (this.state.error) {
//             return <p>{this.state.error.message}</p>;
//         }
//         return this.props.children;
//     }
// }

class App extends React.Component {
    static propTypes = {
        settings: PropTypes.shape({
            locale: PropTypes.string.isRequired,
            fullNode: PropTypes.string.isRequired,
        }).isRequired,
        app: PropTypes.shape({
            isOnboardingCompleted: PropTypes.bool.isRequired,
        }).isRequired,
    };

    state = {
        initialized: false,
    };

    componentWillMount() {
        persistStore(store, { blacklist: ['tempAccount', 'notifications', 'seeds'] }, () => {
            // setTimeout(
            //     () =>
            //         this.setState(() => ({
            //             initialized: true,
            //         })),
            //     2500,
            // );
            // TODO: re-add timeout to avoid flashes of the loading spinner. temporarily disabled for easier debugging
            this.setState(() => ({
                initialized: true,
            }));
        });
    }

    // TODO: this is not working on windows. Investigate why
    // componentDidMount() {
    //     const { remote } = require('electron');
    //     const keytar = remote.require('keytar');
    //     // const keytar = require('keytar');
    //     keytar
    //         .setPassword('iotaWallet', 'Main Wallet', 'AAABBBCCCC999')
    //         .then(() => {
    //             console.log('SAVED SEED');
    //         })
    //         .catch(err => {
    //             console.log('ERROR WHILE SAVING', err);
    //         });
    // }

    componentWillReceiveProps(nextProps) {
        if (nextProps.settings.locale !== this.props.settings.locale) {
            i18next.changeLanguage(nextProps.settings.locale);
        }
    }

    componentDidCatch(error) {
        this.setState(() => ({
            error,
        }));
    }

    render() {
        const { app } = this.props;

        if (this.state.initialized === false) {
            return <Loading />;
        }

        return (
            <div>
                {this.state.error && <p>{this.state.error.message}</p>}
                <Notifications />
                {app.isOnboardingCompleted ? <Main /> : <Onboarding />}
            </div>
        );
    }
}

const mapStateToProps = state => ({
    settings: state.settings,
    app: state.app,
});

export default withRouter(connect(mapStateToProps)(App));
