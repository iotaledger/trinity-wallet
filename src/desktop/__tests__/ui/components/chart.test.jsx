import React from 'react';
import { shallow } from 'enzyme';
import { LineChart } from 'recharts';

import { ChartComponent as Chart } from 'ui/components/Chart';
import defaultTheme from 'themes/themes/Default';

const props = {
    priceData: {
        currency: 'USD',
        globalSymbol: '$',
        symbol: '$',
        price: 1,
        volume: '100',
        change24h: '3',
        mcap: '1000000',
        rates: {}
    },
    chartData: {
        data: [{ x: 0, y: 0, time: 0 }],
        timeframe: '24h',
        yAxis: { ticks: [0] },
    },
    history: {
        push: jest.fn(),
    },
    setCurrency: jest.fn(),
    setTimeframe: jest.fn(),
    getPriceForCurrency: (str) => str,
    getPriceFormat: (str) => str,
    theme: defaultTheme,
    t: (str) => str,
};

describe('Chart component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Chart {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Chart currency title', () => {
        const wrapper = shallow(<Chart {...props} />);

        expect(wrapper.find('h3').text()).toEqual('MIOTA/USD');
    });

    test('Render chart component', () => {
        const wrapper = shallow(<Chart {...props} />);

        expect(wrapper.find(LineChart)).toHaveLength(1);
    });

    test('Chart alter controls', () => {
        const wrapper = shallow(<Chart {...props} />);

        const buttonItems = wrapper.find('Button');
        const callbacks = [props.setCurrency, props.setTimeframe];

        buttonItems.forEach((item, index) => {
            item.simulate('click');
            expect(callbacks[index]).toHaveBeenCalledTimes(1);
        });
    });

    test('Market data stats', () => {
        const wrapper = shallow(<Chart {...props} />);

        const listItems = wrapper.find('ul li');
        const listValues = ['chart:mcap: $ 1000000', 'chart:currentValue: $ USD / Mi', 'chart:change: 3%'];

        listItems.forEach((item, index) => {
            expect(item.text()).toEqual(listValues[index]);
        });
    });
});
