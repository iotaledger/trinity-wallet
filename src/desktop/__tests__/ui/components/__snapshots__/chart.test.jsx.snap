// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart component Render the component 1`] = `
<div
  className="chart"
>
  <h3>
    MIOTA/
    USD
  </h3>
  <div>
    <ResponsiveContainer
      debounce={0}
      height="100%"
      width="100%"
    >
      <LineChart
        barCategoryGap="10%"
        barGap={4}
        data={
          Array [
            Object {
              "time": 0,
              "x": 0,
              "y": 0,
            },
          ]
        }
        layout="horizontal"
        margin={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        reverseStackOrder={false}
        stackOffset="none"
      >
        <Line
          activeDot={true}
          animateNewValues={true}
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          connectNulls={false}
          dataKey="y"
          dot={false}
          fill="#fff"
          hide={false}
          isAnimationActive={false}
          legendType="line"
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          points={Array []}
          stroke="rgb(65, 220, 243)"
          strokeWidth={2}
          type="natural"
          xAxisId={0}
          yAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              "dataMin",
              "dataMax",
            ]
          }
          height={0}
          hide={false}
          interval={0}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          strokeWidth={0}
          tick={
            Object {
              "dx": -56,
              "fill": "rgb(240, 242, 246)",
              "textAnchor": "start",
            }
          }
          tickCount={6}
          tickFormatter={[Function]}
          ticks={
            Array [
              0,
            ]
          }
          type="number"
          width={80}
          yAxisId={0}
        />
        <Tooltip
          active={false}
          allowEscapeViewBox={
            Object {
              "x": false,
              "y": false,
            }
          }
          animationDuration={400}
          animationEasing="ease"
          content={[Function]}
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemStyle={Object {}}
          labelStyle={Object {}}
          maxItems={1}
          offset={10}
          separator=" : "
          symbol="$"
          timeframe="24h"
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
      </LineChart>
    </ResponsiveContainer>
  </div>
  <footer>
    <nav>
      <Button
        className="outline"
        onClick={[Function]}
        variant="secondary"
      >
        USD
      </Button>
      <Button
        className="outline"
        onClick={[Function]}
        variant="secondary"
      >
        24h
      </Button>
    </nav>
    <ul>
      <li>
        <strong>
          chart:mcap
          : 
        </strong>
        $ 1000000
      </li>
      <li>
        <strong>
          chart:currentValue
          : 
        </strong>
        $
         
        USD
         / Mi
      </li>
      <li>
        <strong>
          chart:change
          :
        </strong>
         
        3
        %
      </li>
    </ul>
  </footer>
</div>
`;
